ios/CodePush/CodePush.m

```diff
@@ -458,11 +459,33 @@ - (void)savePendingUpdate:(NSString *)packageHash
                            rejecter:(RCTPromiseRejectBlock)reject)
 {
     NSError *error;
+    if (isRunningBinaryVersion) {
+        // isRunningBinaryVersion will not get set to "YES" if running against the packager.
+        NSString *binaryHash = [CodePushUpdateUtils getHashForBinaryContents:[CodePush binaryBundleURL] error:&error];
+        if (error) {
+            NSLog(@"Error obtaining hash for binary contents: %@", error);
+            resolve(nil);
```

Comment:
Actually, I wonder if it wouldn't make more sense for getConfigiration to return the binary hash and update checkForUpdate to detect a null current package and use the config' hash and version instead. This seems to provided a cleaner seperate on between getCurrentPackage and getConfigiration. Since we don't document getConfiguration, it also allows us to add this new behavior without having to modify the public JS API.